[DEVI-1570] Updates PagerDuty App docs for Scoped OAuth (#68)
PagerDuty/developer-docs: docs/app-integration-development/12-App-OAuth-Token.md

> require a feature flag before this functionality is available to you. Please reach out to us if you have any questions or
> need support.

<!-- theme:warning -->
> ### This flow is only for server-side apps with Scoped OAuth functionality
> A client_secret should be treated as a password and stored securely. It should never be stored in public code.

## About App OAuth Tokens

> To act on a PagerDuty Account as a PagerDuty App, you must obtain an app token for that account. A token is only issued if the application has been previously granted access to that account. Applications are implictly granted access to the account that created it.

Reviewer: ```suggestion
To act on a PagerDuty Account as a PagerDuty App, you must obtain an app token for that account. A token can only be issued against the account that created it.
```

Since there's no publish process just yet.
